waiter/integration/waiter/grpc_test.clj

```diff
@@ -1095,3 +1095,33 @@
                     (assert-grpc-cancel-status status "Cancelled by server" assertion-message)
                     (is (nil? message-summary) assertion-message)
                     (assert-request-state grpc-client request-headers service-id correlation-id ::server-cancel)))))))))))
+
+(deftest ^:parallel ^:integration-fast test-kubernetes-reverse-proxy-sidecar-grpc
+  (testing-using-waiter-url
+    (when (using-k8s? waiter-url)
+      (if-not (contains? (get-kubernetes-scheduler-settings waiter-url) :reverse-proxy)
+        (log/warn "skipping the integration test as :reverse-proxy is not defined")
+        (let [reverse-proxy-flag "REVERSE_PROXY"
+              request-headers (assoc (basic-grpc-service-parameters)
+                                     "content-type" "application/grpc"
+                                     (str "x-waiter-env-" reverse-proxy-flag) "yes")
+              _ (log/info "making canary request")
+              ;; grpc post-data = {id: "x", from: "y", message: "z"}
+              ;; see the Proto3 Language Guide for details on the binary encoding:
+              ;; https://developers.google.com/protocol-buffers/docs/proto3
+              post-data (byte-array [0x00 0x00 0x00 0x00 0x09 0x0a 0x01 0x78 0x12 0x01 0x79 0x1a 0x01 0x7a])
+              {:keys [cookies service-id] :as response} (make-request-with-debug-info
+                                                          request-headers
+                                                          #(make-shell-request waiter-url % :body post-data :path "/courier.Courier/SendPackage"))]
```

Comment:
Nitpick: should we use `start-courier-instance` instead to start the service and then make grpc requests.

Reply:
Fixed